### PR TITLE
Math magicians

### PR DESCRIPTION
**Pull request by gasb150**
In this branch the project was:
1. Refactored: 

- The app is using a Functional component in its totality.
- Add Router to move through tree pages: Home, Calculator, and Quote.
- Add styling.
2. Testing: 

- Setup three types of tests in the project: Unit test for calculate.js and operate.js , snapshot test for App, Button, ButtonPanel, Display, Home and Quote component, and, DOM testing to Home, Calculator and Quote page.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,8 +1,11 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
-import App from './components/App';
+// import App from './components/App';
```
Comment (bellom) [inline src/index.js]: It’s better to delete commented code. If you need to review a previous version of your code, use git history.

File: src/logic/calculate.js [hunk 2]
```diff
@@ -72,8 +73,7 @@ const operationButton = (data, buttonName) => {
       operation = buttonName;
     }
   }
-
   return { total, next, operation };
 };
-const calculate = (data, buttonName) => operationButton(data, buttonName);
+// const calculate = (data, buttonName) => operationButton(data, buttonName);
```
Comment (bellom) [inline src/logic/calculate.js]: It’s better to delete commented code. If you need to review a previous version of your code, use git history.

File: src/__test__/logic/calculate.test.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    expect(result).toStrictEqual({ next: null, operation: null, total: '0.' });
+  });
+  it('Should return a negative value', () => {
+    const result = Calculate(secondObject, '+/-');
+    expect(result).toStrictEqual({ next: -0, operation: null, total: -6 });
+  });
+  it('Should return a percent value', () => {
+    const result = Calculate(secondObject, '%');
+    expect(result).toStrictEqual({ next: 0, operation: null, total: 0.06 });
+  });
```
Comment (bellom) [inline src/__test__/logic/calculate.test.js:37]: Let's add more test cases.

As said in the requirements we also need to test the `calculate.js` functionalities. We need to write tests to verify if each operation (addition, subtraction, multiplication...) and buttons are working as expected.

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @gasb150  :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well adding some test cases, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 

# Status: Changes Required! ♻️ 

**1. Kindly check out my comments below;**
**2.** Kindly fix the failing test.
image



### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Review by arslanbisharat — APPROVED** [review]
# Approved :clap:

Hello Developer :wave:

Thank You for submitting this Pull Request. I am thrilled to inform you that your project is approved now. :clap:

You have done a great job. I like your project. 

**To highlight** 

- [ ] You have a descriptive Readme. Nice Job. 
- [ ] Your Pull Request is nice. Good Job!
- [ ] You have deployed your website. Nice!!!
- [ ] You are using linters and no error is present. I am happy to see it. 
- [ ] Your project has met the minimum technical requirements too. 



In short, I am very excited to see this project and proud of you. 

**What to do next?** :thinking:

1- Visit https://dashboard.microverse.org/projects and submit your approved project. 
2- Prepare yourself for the next project. 

I am wishing you very good luck with the next project. :smile:

Once again, Congratulations!:joy:

**Best,
Arslan Bisharat(PT-TSE)
Slack: @arslanbisharat**

**Pull request merged**